src/main/java/uk/gov/hmcts/reform/prl/services/ManageOrderEmailService.java

```diff
@@ -92,16 +89,64 @@ private List<PartyDetails> getApplicants(CaseData caseData) {
 
 
     private List<String> getEmailAddress(List<Element<PartyDetails>> partyDetails) {
-        List<String> emails = partyDetails
+        return partyDetails
+            .stream()
+            .map(Element::getValue)
+                .filter(a -> a.getCanYouProvideEmailAddress().equals(YesOrNo.Yes))
+            .map(PartyDetails::getEmail)
+            .collect(Collectors.toList());
+    }
+
+
+    public void sendEmailToCafcass(CaseDetails caseDetails) {
+
+        log.info("We are about to send an email to cafcass");
+
+        CaseData caseData = emailService.getCaseData(caseDetails);
+
+        ManageOrders manageOrders  = caseData.getManageOrders();
+
+        List<String> cafcassEmails = manageOrders.getCafcassEmailAddress()
             .stream()
-            .map(Element::getValue).filter(partyDetails1 -> true).map(element -> element.getEmail())
+            .map(Element::getValue)
             .collect(Collectors.toList());
 
+        log.info("Cafcass email id {}", cafcassEmails);
 
-        emails.removeIf(Objects::isNull);
+        cafcassEmails.forEach(email ->   emailService.send(
+            email,
+            EmailTemplateNames.CAFCASS,
+            buildCafcassEmail(caseDetails),
```

Comment:
Can we pass 'caseDate' as method parameter instead of 'caseDetails' to avoid extraction again within method?

Reply:
ok, I will update this as well